examples/refsi/hal_refsi/include/device/program.lds

```diff
@@ -17,57 +17,57 @@
 
 OUTPUT_ARCH( "riscv" )
 
+MEMORY
+{
+    mainmem : ORIGIN = 0x10000, LENGTH = (1 << 20)
```

Comment:
It seems like it's possible to mix explicit regions and implicit regions (looking at GNU ld's own linker scripts, which do that), so we could potentially drop `mainmem`. But I don't mind if you want to keep that in.

Reply:
I think it's useful to keep in - If we try to make a binary larger than `1 << 20` bytes, then the linker will fail with an appropriate error message rather than having the program loader crash.

Reply:
That makes sense to me, thanks.